File: POProjekt3/rosliny/Jagody.py
from Roslina import Roslina

class Jagody(Roslina):
    
    def __init__(self):
        pass

    def __init__(self, swiat, poz, wiek, inicj, moc):

        super(Roslina, self).__init__()

        super().changeSwiat(swiat)
        super().changePolozenie(poz[0], poz[1])
        super().changeWiek(wiek)
        super().changeInicjatywa(inicj)

        s = "J"
        super().getSwiat().edytujPlansze(poz[0], poz[1], s)

    def __init__(self, swiat, poz):

        super(Roslina, self).__init__()

        super().changeSwiat(swiat)
        super().changePolozenie(poz[0], poz[1])
        super().changeInicjatywa(0)
        super().zmienSile(99)
        super().changeWiek(0)

        s = "J"
        super().getSwiat().edytujPlansze(poz[0], poz[1], s)

    #--------------------------------------------

    def rysowanie(self):
        print('J')

    def getZnak(self):
        return 'J'

    def czyZabojcza(self):
        return True

File: VirtualWorld-Java/src/Jagody.java

import java.util.ArrayList;


public class Jagody extends Roslina{
	
	public Jagody() {
		
	}
	
	public Jagody(Swiat swiat, ArrayList<Integer> poz, int wiek, int inicj, int moc) {
		changeSwiat(swiat);
		changePolozenie(poz.get(0), poz.get(1));
		changeWiek(wiek);
		changeInicjatywa(inicj);
		
		String s = ""+'J';
		getSwiat().edytujPlansze(poz.get(0), poz.get(1), s);
	}
	
	public Jagody(Swiat swiat, ArrayList<Integer> poz) {
		changeSwiat(swiat);
		changePolozenie(poz.get(0), poz.get(1));
		changeInicjatywa(0);
		zmienSile(99);
		changeWiek(0);
		
		String s = ""+'J';
		getSwiat().edytujPlansze(poz.get(0), poz.get(1), s);
	}
	
	@Override
	public void rysowanie() {
		System.out.println('J');
	}
	
	@Override
	public char getZnak() {
		return 'J';
	}
	
	@Override
	public Boolean czyZabojcza() {
		return true;
	}
	
}
